qiita_db/sql_connection.py

```diff
@@ -22,57 +22,101 @@
-Modifying the previous example to show this ability (Note the RETURNING added
-to the first SQL command):
-
-from qiita_db.sql_connection import SQLConnectionHandler
-conn_handler = SQLConnectionHandler # doctest: +SKIP
-conn_handler.create_queue("example_queue") # doctest: +SKIP
-conn_handler.add_to_queue(
-    "example_queue", "INSERT INTO qiita.qiita_user (email, name, password,"
-    "phone) VALUES (%s, %s, %s, %s) RETURNING email, password",
-    ['insert@foo.bar', 'Toy', 'pass', '111-111-11112']) # doctest: +SKIP
-conn_handler.add_to_queue(
-    "example_queue", "UPDATE qiita.qiita_user SET user_level_id = 1, "
-    "phone = '222-222-2221' WHERE email = %s AND password = %s",
-    ['{0}', '{1}']) # doctest: +SKIP
-conn_handler.execute_queue("example_queue") # doctest: +SKIP
-conn_handler.execute_fetchall(
-    "SELECT * from qiita.qiita_user WHERE email = %s", ['insert@foo.bar'])
-[['insert@foo.bar', 1, 'pass', 'Toy', None, None, '222-222-2221', None, None,
-  None]] # doctest: +SKIP
+
+>>> from qiita_db.sql_connection import TRN
+
+Transaction blocks are used through the variable TRN available in this module.
+TRN should be used as a context manager. You can add as many SQL commands as
+you want and execute all of them at once, and it will return the results of
+all the commands.
+
+>>> with TRN:
+...     TRN.add("SELECT 42")
+...     TRN.add("SELECT 43")
+...     res = TRN.execute()  # The transaction is not committed here
+>>> # The transaction committed here
+>>> res
+[[[42]], [[43]]]
+
+You can have nested transactions and they will not commit until the first
+transaction (represented by entering to the context) is exited:
+
+>>> with TRN:
+...     TRN.add("SELECT 42")
+...     with TRN:
+...         TRN.add("SELECT 43")
+...         res = TRN.execute()
+...     # The transaction is still not committed
+...     TRN.add("SELECT 44")
+...     res = TRN.execute()
+>>> # The transactions committed here
+>>> res
+[[[42]], [[43]], [[44]]]
+
+You can also use results from a previous command in the transaction in a later
+command without partially executing the transaction by using placeholders. The
+placeholders are of the form {int:int:int}, where the first integer is the
+index of the query in the transaction, the second integer is the row in the
+query results where the value exists, and the last integer is the actual index
+of the desired value in the result row.
+
+>>> with TRN:
+...     TRN.add("SELECT 42")
+...     TRN.add("SELECT 42 + %s", ["{0:0:0}"])
+...     res = TRN.execute()
+>>> res
+[[[42]], [[84]]]
+
+If you don't know the index of the query because your current transaction can
+be embedded in a larger transaction, use the function `TRN.index` before adding
+the target query:
+
+>>> with TRN:
+...     idx = TRN.index
+...     TRN.add("SELECT 42")
+...     TRN.add("SELECT 42 + %s", ["{%d:0:0}" % idx])
+...     res = TRN.execute()
+>>> res
+[[[42]], [[84]]]
+
+Multiple auxiliary functions exist for easy SQL result retrieval:
```

Comment:
I don't think we can add headers per se in here, as the headers are a controlled set in the documentation (e.g. here the header is "Examples"). However I can try to add a unordered list...